service/apis/qsl_service.go

```diff
@@ -0,0 +1,379 @@
+func (qa *QSLService) CreateDgraphQueryHelper(query []string, tabs int, parent string) ([]string, error) {
+
+	// string that we're going to replace
+	basequery := []string{
+		strings.Repeat("\t", tabs) + "$RELATION @filter(eq(objtype, $OBJTYPE) and$FILTERSFUNC){",
+	}
+	_ = basequery
+
+	// regex to match the string pattern
+	//r := regexp.MustCompile(`([a-zA-Z]+)\[(\@[\,\@\"\=a-zA-Z0-9\-\.\|]*)\]\{([\*|[\,\@\"\=a-zA-Z0-9\-]*)`)
+	//r := regexp.MustCompile(`([a-zA-Z]+)\[(?:(\@[\,\@\"\=a-zA-Z0-9\-\.\|\:_]*|\*))\]\{([\*|[\,\@\"\=a-zA-Z0-9\-]*)`)
+	r := regexp.MustCompile(blockRegex)
+	matches := r.FindStringSubmatch(query[0])
+	// log.Infof("%#v\n", splitquery)
+	log.Debugf("helpermatches %#v\n", matches)
+
+	// extract the values of the form objtype[filters]fields and assign to individual variables
+	objtype := strings.Title(matches[1])
+	filters := matches[2]
+	fields := matches[3]
+
+	// get a list of the metadata fields for this object type
+	metafieldslist, err := qa.metaSvc.GetMetadataFields(objtype)
+	if err != nil {
+		log.Error("err in getting metadata fields for " + objtype)
+		log.Error(err)
+		return nil, errors.New("Failed to connect to dgraph to get metadata")
+	}
+
+	if len(metafieldslist) == 0 {
+		log.Error("metadata for " + objtype + " not found in db. Will not be able to use * or find relationships")
+	}
+
+	log.Debugf("metadata fields for %s: %#v", objtype, metafieldslist)
+
+	// declare relation variable
+	relation := ""
+	// see if we can find the reverse relation from this object to its parent
+	found := false
+	// look in the list of fields for the metadata and
+	// find if there's a relationship between the parent's and this object's type
+	// e.g. if we had cluster[...]{...}.pod[...]{...} parent=cluster
+	// and we will find the pods relation to cluster is called ~cluster
+	for i, item := range metafieldslist {
+		_ = i
+
+		if item.FieldType == "relationship" {
+			log.Debugf("1 found relationship for %s-%s->%s", objtype, item.FieldName, item.RefDataType)
+			if item.RefDataType == strings.Title(parent) {
+				relation = "~" + strings.ToLower(item.FieldName)
+				found = true
+			}
+		}
+	}
+
+	if !found {
+		// if not, see if we can find the relation from the parent to this object
+		metafieldslist2, err := qa.metaSvc.GetMetadataFields(parent)
+		if err != nil {
+			log.Error("err in getting metadata fields")
+			log.Error(err)
+			return nil, errors.New("Failed to connect to dgraph to get metadata")
+		}
+		log.Debugf("couldn't find relation for %s->%s,", parent, objtype)
+		log.Debugf("metadata fields for %s: %#v", parent, metafieldslist)
+
+		for i, item := range metafieldslist2 {
+			_ = i
+			if item.FieldType == "relationship" {
+				log.Debugf("2 found relationship for %s-%s->%s", parent, item.FieldName, item.RefDataType)
+				if item.RefDataType == strings.Title(objtype) {
+					relation = strings.ToLower(item.FieldName)
+					found = true
+				}
+			}
+		}
+	}
+
+	// no relation found between the two objects
```

Comment:
@kianjones4, maybe highlight the word "still" in this comment, ex: `//still no relation found...` or some other change to indicate why there are two `if !found {...` blocks in a row... which may be otherwise confusing initially to the future reader